pkg/machines/components/vmSnapshotsTab.jsx

```diff
@@ -43,118 +44,153 @@ function prettyTime(unixTime) {
-            const columns = detailMap.map(d => {
-                let column = null;
-                if (typeof d.value === 'string') {
-                    if (target[d.value] !== undefined)
-                        column = { title: <div id={`${id}-snapshot-${snapId}-${d.value}`}>{target[d.value]}</div> };
-                }
-                if (typeof d.value === 'function')
-                    column = { title: d.value(target, snapId) };
+class VmSnapshotsTab extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showCreateSnapshotModal: false,
+        };
 
-                return column;
-            });
-            return { columns };
-        });
+        this.openCreateSnapshot = this.openCreateSnapshot.bind(this);
+        this.closeCreateSnapshot = this.closeCreateSnapshot.bind(this);
+    }
+
+    openCreateSnapshot() {
+        this.setState({ showCreateSnapshotModal: true });
     }
 
-    return (
-        <ListingTable aria-label={`VM ${vm.name} Snapshots Cards`}
-            variant='compact'
-            emptyCaption={emptyCaption}
-            columns={columnTitles}
-            rows={rows} />
-    );
-};
+    closeCreateSnapshot() {
+        this.setState({ showCreateSnapshotModal: false });
+    }
+
+    render() {
+        const { vm, dispatch } = this.props;
+        const id = vmId(vm.name);
+
+        const emptyCaption = (<div className="no-snapshots">
+            <h3>{_("No snapshots")}</h3>
+            {_("Previously taken snapshots allow you to revert to an earlier state if something goes wrong")}
+        </div>);
+
+        let detailMap = [
+            {
+                name: _("Creation Time"), value: (snap, snapId) => {
+                    const date = prettyTime(snap.creationTime);
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-date`}>
+                            {date}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("Name"), value: (snap, snapId) => {
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-name`}>
+                            {snap.name}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("Description"), value: (snap, snapId) => {
+                    let desc = snap.description;
+                    if (!desc)
+                        desc = (<span className="snap-greyed-out">{_("No description")}</span>);
+
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-description`}>
+                            {desc}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("State"), value: (snap, snapId) => {
```

Comment:
Isn't this change unrelated?

Reply:
oh the changes i just done in previous PR gets overwritten. Thanks for catching that!